agent/algorithm/binary/binary.go

```diff
@@ -73,10 +67,10 @@ func (b *binary) Run() ([]byte, error) {
 		return nil, fmt.Errorf("algorithm execution error: %v", err)
 	}
 
-	select {
-	case result = <-dataChannel:
-		return result, nil
-	case err = <-errorChannel:
-		return nil, fmt.Errorf("error receiving data: %v", err)
+	results, err := algorithm.ZipDirectory()
```

Comment:
should be done by agent, when results are consumed. Cleanup can happen after agent gets to complete state